lms/djangoapps/courseware/grades.py

```diff
@@ -727,3 +737,14 @@ def iterate_grades_for(course_or_id, students, keep_raw_scores=False):
                     exc.message
                 )
                 yield student, {}, exc.message
+
+
+def _get_mock_request(student):
+    """
+    Make a fake request because grading code expects to be able to look at
+    the request. We have to attach the correct user to the request before
+    grading that student.
+    """
+    request = RequestFactory().get('/')
```

Comment:
I think it'll be ok. Grading generates a whole lot of objects with gnarly links between them, but this seems fairly straightforward.